dvc/repo/plots/data.py

```diff
@@ -152,7 +146,7 @@ def _append_revision(data_points, revision, **kwargs):
 
 class PlotData:
     REVISION_FIELD = "rev"
-    INDEX_FIELD = "index"
+    INDEX_FIELD = "step"
```

Comment:
Seems like `step` is generic enough to replace `index` - in both cases they do not make too much sense when plotting anything, so we agreed with @dmpetrov that we can replace it for the sake of "additional config"-less integration with logs.

Reply:
What about backward compatibility?

Reply:
@efiop What do you mean by backward compatibility?
`INDEX_FIELD` is appended dynamically to data if `-x` [option is not provided](https://github.com/iterative/dvc/blob/4cf2f8139bdd2c30d439dea1bb7375ddb63e46d0/dvc/repo/plots/data.py#L132). So, if the question is whether we will break repositories already using `plots`? No, we will not, we will just replace `index` with `step` in the actual plot.
But if you mean whether default behavior of plots will change, then yes, since the default `x` axis name changes.